### ISSUE TITLE
matchtask rules don't work in the nested blocks

### ISSUE THREAD

**Issue by aliaksandr-dounar-epam** [ansible/ansible-lint#527]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version
ansible 2.7.11
ansible-lint --version
ansible-lint 4.1.0
```

- ansible installation method: pip
- ansible-lint installation method: pip
# Desired Behaviour

Ansible-lint rules for tasks validation work correctly even under the nested blocks.
When I remove top-level block from the example below everything works as expected.

Example playbook:
```yaml
- name: 'playbook for task category'
  gather_facts: false
  tasks:
    - block:
      - block:
          - package:
              name: '{{ package_install_name }}'
              state: '{{ package_install_state }}'
              enablerepo: '{{ package_repo }}'
            loop:
              - 'one'
              - 'two'
            become: true
            when: 'some_var == "Value"'
        when: 'not package_from_s3'
```

# Actual Behaviour (Bug report only)
```bash
ansible-lint tests/negative/test.yml
echo $?
0
echo "removing nested block"
ansible-lint tests/negative/test.yml
[502] All tasks should be named
tests/negative/packages-dont-have-retry.yml:7
```

**Comment by tgadiev** [comment]
Might be similar to https://github.com/ansible/ansible-lint/issues/355

**Comment by ssbarnea** [comment]
Closing as not being able to reproduce with v6.
